STACK-UP-3/champs-backend
# Adding Winston & Connected Logdna

#### Description
Added Winston for logging and connected logdna as an API that can be used for logs retrieval, and analysis.
#### Type of change

Please select the relevant option


#### How Has This Been Tested?


#### How to Test

Give instructions on how reviewers can verify your work

- Run `npm run serve `

- There are 2 endpoints you can access to verify added middleware `bnLog` that works;
  [`get <hostname>/success`](http://localhost:3000/success) logs when the function runs successfully without any error,
  [`get <hostname>/error`](http://localhost:3000/error) logs when the function fail with an error,

- Open [Logdna](https://app.logdna.com/006e982ec8/logs/view/ed75f2bc33?hosts=shama-machine) to retrieve all logs.
  _Username: irutingabos3@gmail.com_
  _password: passwordme_

#### Any background context you want to add (Optional)
You can use  `bnLog` function as middleware on your function to log the results like errors.
example: 
`const dumFunction = (req, res) => { res.send('Working well!'); };`
`app.get('/test', bnLog(dumFunction))
`

Screenshots or GIFs
![image](https://user-images.githubusercontent.com/11549858/77657606-34c35380-6f7e-11ea-8439-a970726720b3.png)


#### STORY ID
[#171912819](https://www.pivotaltracker.com/story/show/171912819)
[Finishes [#171912819](https://www.pivotaltracker.com/story/show/171912819)]
